pkg/registry-disk/registry-disk.go

```diff
@@ -81,6 +82,28 @@ func getFilePath(basePath string) (string, string, error) {
 	return "", "", errors.New(fmt.Sprintf("no supported file disk found in directory %s", basePath))
 }
 
+func CleanupOrphanedEphemeralDisks(indexer cache.Store) error {
+	vms, err := diskutils.ListVmWithEphemeralDisk(mountBaseDir)
+	if err != nil {
+		return err
+	}
+
+	for _, vm := range vms {
+		key, err := cache.MetaNamespaceKeyFunc(vm)
+		if err != nil {
+			return err
+		}
+		_, exists, _ := indexer.GetByKey(key)
```

Comment:
I think you also want to clean it up, if it is in a succeeded or failed state.

Reply:
good catch